tests/regression/phenology/test_phenology_gpp.py

```diff
@@ -74,17 +75,30 @@ def test_phenology_gpp_calculation(
     # - PMod_sub_A0_daily_total
     # - PMod_sub_A0_daily_total_penalised
 
-    hh_values = pd.DataFrame(
-        dict(
-            time=de_gri_subdaily_data["time"],
-            gpp=de_gri_subdaily_pmodel.gpp,
-        ),
-    )
-    hh_values = hh_values.set_index("time")
-    hh_resampler = hh_values.resample("D")
+    daily_gpp = de_gri_subdaily_pmodel._get_daily_gpp()
 
     assert_allclose(
         de_gri_daily_outputs["PMod_gpp_smstress"],
-        hh_resampler["gpp"].mean() * soilm_stress,
+        daily_gpp * soilm_stress,
+        rtol=1e-6,
+    )
+
+    fortnightly_env = PModelEnvironment(
+        tc=de_gri_fortnightly_data["tc_mean"].to_numpy(),
+        vpd=de_gri_fortnightly_data["vpd_mean"].to_numpy(),
+        patm=de_gri_fortnightly_data["patm_mean"].to_numpy(),
+        co2=de_gri_fortnightly_data["co2_mean"].to_numpy(),
+        ppfd=de_gri_fortnightly_data["ppfd_mean"].to_numpy(),
+        fapar=np.ones_like(de_gri_fortnightly_data["tc_mean"]),
+    )
+    fortnightly_datetimes = de_gri_fortnightly_data["time"]
+    de_gri_pmodel = PModel(env=fortnightly_env)
+    pmodel_gpp_from_fortnightly = de_gri_pmodel._get_daily_gpp(
+        datetimes=fortnightly_datetimes
+    )
+
+    assert_allclose(
+        de_gri_fortnightly_daily_outputs["daily_gpp"],
+        pmodel_gpp_from_fortnightly[:-1],
```

Comment:
I am not sure why this is needed. The arrays seem to be off.

Reply:
It tells me the arrays are off by one if I don't add it, but still tells me the same if I use `[:-2]`